tools/ci_build/github/azure-pipelines/templates/mac-cpu-packaging-pipeline.yml

```diff
@@ -22,78 +22,103 @@ parameters:
         WithCache: ${{ parameters.WithCache }}
-    - template: mac-cpu-packing-jobs.yml
-      parameters:
-        MacosArch: 'arm64'
-        AllowReleasedOpsetOnly: ${{ parameters.AllowReleasedOpsetOnly }}
-        AdditionalBuildFlags: ${{ parameters.AdditionalBuildFlags }}
-        WithCache: ${{ parameters.WithCache }}
-    - template: mac-cpu-packing-jobs.yml
-      parameters:
-        MacosArch: 'universal2'
-        AllowReleasedOpsetOnly: ${{ parameters.AllowReleasedOpsetOnly }}
-        AdditionalBuildFlags: ${{ parameters.AdditionalBuildFlags }}
-        WithCache: ${{ parameters.WithCache }}
-    - job: MacOS_C_API_Package_Publish_All
-      dependsOn:
-      - MacOS_C_API_Packaging_CPU_x86_64
-      - MacOS_C_API_Packaging_CPU_arm64
-      - MacOS_C_API_Packaging_CPU_universal2
-      pool:
-        vmImage: 'macOS-12'
-      steps:
-      - task: DownloadPipelineArtifact@2
-        inputs:
-          artifact: 'onnxruntime-osx-x86_64'
-          targetPath: '$(Build.ArtifactStagingDirectory)'
-      - task: DownloadPipelineArtifact@2
-        inputs:
-          artifact: 'onnxruntime-osx-arm64'
-          targetPath: '$(Build.ArtifactStagingDirectory)'
-      - task: DownloadPipelineArtifact@2
-        inputs:
-          artifact: 'onnxruntime-osx-universal2'
-          targetPath: '$(Build.ArtifactStagingDirectory)'
-      - task: PublishPipelineArtifact@1
-        inputs:
-          targetPath: '$(Build.ArtifactStagingDirectory)'
-          artifact: 'onnxruntime-osx'
-          condition: 'succeededOrFailed()'
-      - template: component-governance-component-detection-steps.yml
+
+    - ${{ if eq(parameters.BuildForAllArchs, true) }}:
+      - template: mac-cpu-packing-jobs.yml
         parameters:
-          condition: 'succeeded'
-  - ${{ if ne(parameters.BuildForAllArchs, true) }}:
-    - template: mac-cpu-packing-jobs.yml
-      parameters:
-        MacosArch: 'x86_64'
-        AllowReleasedOpsetOnly: ${{ parameters.AllowReleasedOpsetOnly }}
-        AdditionalBuildFlags: ${{ parameters.AdditionalBuildFlags }}
-        WithCache: ${{ parameters.WithCache }}
-    - job: MacOS_C_API_Package_Publish_x86_64
-      dependsOn: MacOS_C_API_Packaging_CPU_x86_64
+          MacosArch: 'arm64'
+          AllowReleasedOpsetOnly: ${{ parameters.AllowReleasedOpsetOnly }}
+          AdditionalBuildFlags: ${{ parameters.AdditionalBuildFlags }}
+          WithCache: ${{ parameters.WithCache }}
+      - template: mac-cpu-packing-jobs.yml
+        parameters:
+          MacosArch: 'universal2'
+          AllowReleasedOpsetOnly: ${{ parameters.AllowReleasedOpsetOnly }}
+          AdditionalBuildFlags: ${{ parameters.AdditionalBuildFlags }}
+          WithCache: ${{ parameters.WithCache }}
+
+- stage: MacOS_C_API_Package_Publish
+  dependsOn: MacOS_C_API_Packaging_CPU
+  jobs:
+    - job: MacOS_C_API_Package_Publish
       pool:
-        vmImage: 'macOS-12'
+        ${{ if eq(parameters.DoESRP, true)}}:
```

Comment:
This 
```
        ${{ if eq(parameters.DoESRP, true)}}:
          vmImage: 'macOS-12'
        ${{ else }}:
          vmImage: 'macOS-13'
```
Should be changed back to 
```
          vmImage: 'macOS-12'
```
since the 1.15 release was built on 'macOS-12'. If you used the newer version, some parts may fail.

Reply:
Though it only use codesign command in this job. I'll remove it.